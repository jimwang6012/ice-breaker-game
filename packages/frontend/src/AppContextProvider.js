import React from "react";
import { useListState } from "@mantine/hooks";
import { useState } from "react";
import useAudio from "./hook/useAudio";

export const AppContext = React.createContext({
  players: [],
  handlers: null,
  isHost: false,
  setIsHost: (isHost) => {},
  roomId: "",
  setRoomId: (roomId) => {},
  hostId: "",
  setHostId: (hostId) => {},
  name: "",
  setName: (name) => {},
  isTyping: false,
  setIsTyping: (isTyping) => {},
  config: {
    roomSize: 8,
    boardSize: 9,
    roundTime: 30,
    breakTime: 0.5,
  },
  setConfig: (config) => {},
  colors: [],
  setColors: (colors) => {},
  colorStatus: [],
  setColorStatus: (colorStatus) => {},
  playing: true,
  toggle: () => {},
  stop: () => {},
});

export function AppContextProvider({ children }) {
  // Manage players in the room
  const [players, handlers] = useListState();

  // Whether the player is a host
  const [isHost, setIsHost] = useState(false);

  // The current room id
  const [roomId, setRoomId] = useState("");

  // The host's id
  const [hostId, setHostId] = useState("");

  //player name
  const [name, setName] = useState("");

  //game config
  const [config, setConfig] = useState({
    roomSize: 8,
    boardSize: 9,
    roundTime: 30,
    breakTime: 0.5,
  });

  //player colors
  const [colors, setColors] = useState([]);

  //player color status
  const [colorStatus, setColorStatus] = useState([]);

  //isChatting
  const [isTyping, setIsTyping] = useState(false);

  const { playing, toggle, stop } = useAudio("../music/music2.ogg");

  const context = {
    players,
    handlers,
    isHost,
    setIsHost,
    roomId,
    setRoomId,
    hostId,
    setHostId,
    name,
    setName,
    config,
    setConfig,
    colors,
    setColors,
    colorStatus,
    setColorStatus,
    isTyping,
    setIsTyping,
    playing,
    toggle,
    stop,
  };

  return <AppContext.Provider value={context}>{children}</AppContext.Provider>;
}
