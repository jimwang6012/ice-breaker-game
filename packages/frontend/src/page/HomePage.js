import React, { useContext, useState } from "react";
import { useNavigate } from "react-router-dom";
import { AppContext } from "../AppContextProvider";
import { MainButton } from "../component/MainButton";

function HomePage() {
  document.body.style.overflow = "hidden";
  const [pin, setPin] = useState("");
  const navigate = useNavigate();
  const { setIsHost, setRoomId } = useContext(AppContext);
  const createGame = () => {
    setIsHost(true);
    navigate("/name");
  };
  const joinGame = () => {
    setIsHost(false);
    setRoomId(pin);
    navigate("/name");
  };

  return (
    <div className="overflow-hidden">
      <div className="flex items-center justify-center h-screen overflow-y-auto bg-ice-8 opacity-90">
        {/* left square */}
        <div className="absolute top-0 rotate-45 -left-96 h-2/3 aspect-square bg-ice-3" />
        <div className="z-40 flex flex-col items-center justify-center gap-20 ">
          <div className="text-6xl font-bold text-white">Ice Breakers!</div>
          {/* center modal */}
          <div className="z-40 py-16 rounded-lg shadow-lg px-28 bg-ice-2">
            <div className="flex flex-row justify-start gap-10">
              <input
                className="justify-center text-xl font-bold text-center border rounded-md text-ice-7 px-14 bg-ice-0 border-ice-6"
                placeholder="Enter the room code"
                onChange={(e) => setPin(e.target.value)}
              />
              <MainButton handleClick={joinGame} text="Join Room" />
            </div>
          </div>
          <MainButton handleClick={createGame} text="Create Game" />
          {/* right square */}
          <div className="absolute bottom-0 rotate-45 -right-96 h-2/3 aspect-square bg-ice-3" />
        </div>
      </div>
    </div>
  );
}

export default HomePage;
