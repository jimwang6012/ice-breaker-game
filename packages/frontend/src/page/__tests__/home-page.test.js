import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { AppContextProvider } from "../../AppContextProvider";
import HomePage from "../HomePage";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <AppContextProvider>
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </AppContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
