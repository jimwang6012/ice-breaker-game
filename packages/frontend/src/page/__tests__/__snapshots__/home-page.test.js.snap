// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="overflow-hidden"
>
  <div
    className="flex items-center justify-center h-screen overflow-y-auto bg-ice-8 opacity-90"
  >
    <div
      className="absolute top-0 rotate-45 -left-96 h-2/3 aspect-square bg-ice-3"
    />
    <div
      className="z-40 flex flex-col items-center justify-center gap-20 "
    >
      <div
        className="text-6xl font-bold text-white"
      >
        Ice Breakers!
      </div>
      <div
        className="z-40 py-16 rounded-lg shadow-lg px-28 bg-ice-2"
      >
        <div
          className="flex flex-row justify-start gap-10"
        >
          <input
            className="justify-center text-xl font-bold text-center border rounded-md text-ice-7 px-14 bg-ice-0 border-ice-6"
            onChange={[Function]}
            placeholder="Enter the room code"
          />
          <div
            className="flex"
          >
            <button
              className="absolute z-40 px-6 py-3 text-xl font-semibold text-white rounded-lg shadow-inner bg-ice-6 hover:mt-1 active:mt-2"
              onClick={[Function]}
            >
              Join Room
            </button>
            <div
              className="px-6 py-3 mt-2 text-xl font-semibold rounded-lg text-ice-7 bg-ice-7"
            >
              Join Room
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex"
      >
        <button
          className="absolute z-40 px-6 py-3 text-xl font-semibold text-white rounded-lg shadow-inner bg-ice-6 hover:mt-1 active:mt-2"
          onClick={[Function]}
        >
          Create Game
        </button>
        <div
          className="px-6 py-3 mt-2 text-xl font-semibold rounded-lg text-ice-7 bg-ice-7"
        >
          Create Game
        </div>
      </div>
      <div
        className="absolute bottom-0 rotate-45 -right-96 h-2/3 aspect-square bg-ice-3"
      />
    </div>
  </div>
</div>
`;
